pkg/server/api/agent/v1/service.go

```diff
@@ -47,7 +55,52 @@ func (s *Service) DeleteAgent(ctx context.Context, req *agent.DeleteAgentRequest
 }
 
 func (s *Service) BanAgent(ctx context.Context, req *agent.BanAgentRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if req.Id == nil {
+		log.Error("Invalid request: ID is nil")
+		return nil, status.Error(codes.InvalidArgument, "ID is nil")
+	}
+
+	id, err := spiffeid.New(req.Id.TrustDomain, req.Id.Path)
```

Comment:
is this conversion common enough to split out into a shared function?

Reply:
This function lives in the go-spiffe library

Reply:
you must use `https://github.com/spiffe/spire/blob/master/pkg/server/api/api.go#L14` here